aa-sdk/core/src/account/smartContractAccount.ts

```diff
@@ -155,7 +157,10 @@ export type ToSmartContractAccountParams<
   // if not provided, will default to just using signMessage over the Hex
   signUserOperationHash?: (uoHash: Hex) => Promise<Hex>;
   encodeUpgradeToAndCall?: (params: UpgradeToAndCallParams) => Promise<Hex>;
-} & Omit<CustomSource, "signTransaction" | "address">;
+} & Omit<
+  CustomSource,
+  "signTransaction" | "address" | "experimental_signAuthorization"
```

Comment:
one caveat here: viem by default will use current nonce to sign the 1559 transaction, and use nonce +1 to sign authorization tuple. if we support 4337, UO is not tracked by native nonce, but by EP's nonce tracking. That means we have to override the default nonce viem used current nonce.

Reply:
https://github.com/wevm/viem/blob/0c41e947ed9174b719bfc83279a307df054f4dce/src/experimental/eip7702/actions/prepareAuthorization.ts#L137